contracts/cw-blob/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use cosmwasm_std::{entry_point, DepsMut, Empty, Env, MessageInfo, Never, Response};
+
+#[entry_point]
+pub fn instantiate(_: DepsMut, _: Env, _: MessageInfo, _: Empty) -> Result<Response, Never> {
+    Ok(Response::new())
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub mod interface {
+    use super::*;
+
+    use cosmwasm_std::{instantiate2_address, Binary, CanonicalAddr, Checksum};
+    use cw_orch::{contract::Contract, prelude::*};
+
+    // We don't want it to be manually instantiated/executed/etc, only uploaded. So not using cw_orch interface
+    #[derive(Clone)]
+    pub struct CwBlob<Chain>(Contract<Chain>);
+
+    impl<Chain> CwBlob<Chain> {
+        pub fn new(contract_id: impl ToString, chain: Chain) -> Self {
+            Self(Contract::new(contract_id, chain))
+        }
+    }
+
+    impl<Chain: ChainState> ContractInstance<Chain> for CwBlob<Chain> {
+        fn as_instance(&self) -> &Contract<Chain> {
+            &self.0
+        }
+
+        fn as_instance_mut(&mut self) -> &mut Contract<Chain> {
+            &mut self.0
+        }
+    }
+
+    impl<T: CwEnv> Uploadable for CwBlob<T> {
+        fn wasm(_chain: &ChainInfoOwned) -> WasmPath {
+            artifacts_dir_from_workspace!()
+                .find_wasm_path("cw_blob")
+                .unwrap()
+        }
+
+        fn wrapper() -> Box<dyn MockContract<Empty, Empty>> {
+            Box::new(
+                ContractWrapper::new_with_empty(
+                    |_, _, _, _: Empty| -> Result<Response, Never> { unreachable!() },
+                    super::instantiate,
+                    |_, _, _: Empty| -> Result<Binary, Never> { unreachable!() },
+                )
+                .with_checksum(Self::checksum()),
+            )
+        }
+    }
+
+    impl<T: CwEnv> CwBlob<T> {
+        /// Checksum of the blob
+        pub fn checksum() -> Checksum {
+            <Self as Uploadable>::wasm(&ChainInfoOwned::default())
+                .checksum()
+                .unwrap()
+        }
+
+        /// Instantiate blob and migrate to your desired contract.
+        /// It will upload your contract, if it's not uploaded already
+        ///
+        /// Checksum of the uploaded blob_code_id on chain should match [CwBlob::checksum()]
+        pub fn upload_and_migrate<M>(
```

Comment:
Can we turn this into a trait? 

I.e. I can have a contract `T` and it implements this method on it but we name it something like `fn deterministic_instantiate` or similar?

Reply:
Like that? Not sure about trait name